src/EditorFeatures/Core/FindUsages/IRemoteFindUsagesService.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.Remote;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.FindUsages
+{
+    internal interface IRemoteFindUsagesService
+    {
+        Task FindReferencesAsync(
+            PinnedSolutionInfo solutionInfo,
+            SerializableSymbolAndProjectId symbolAndProjectIdArg,
+            SerializableFindReferencesSearchOptions options,
+            CancellationToken cancellationToken);
+    }
+
+    internal class FindReferencesServerCallback
+    {
+        private readonly Solution _solution;
+        private readonly IFindUsagesContext _context;
+        private readonly Dictionary<int, DefinitionItem> _idToDefinition = new Dictionary<int, DefinitionItem>();
+
+        public FindReferencesServerCallback(Solution solution, IFindUsagesContext context)
+        {
+            _solution = solution;
+            _context = context;
+        }
+
+        public Task AddItemsAsync(int count)
+            => _context.ProgressTracker.AddItemsAsync(count);
+
+        public Task ItemCompletedAsync()
+            => _context.ProgressTracker.ItemCompletedAsync();
+
+        public Task ReportMessageAsync(string message)
+            => _context.ReportMessageAsync(message);
+
+        [Obsolete]
```

Comment:
I see this same method obsoleted in other places too. Can this one have a message too?

Reply:
yup.  i will add :)